src/number_format.js

```diff
@@ -277,8 +282,18 @@ class NumberFormat extends React.Component {
 
     //in case of format as number limit between prefix and suffix
     if (!format) {
-      const hasNegation = value[0] === '-';
-      return clamp(caretPos, prefix.length + (hasNegation ? 1 : 0), value.length - suffix.length);
+      let hasNegation;
+      let negationLength = 0;
+      if (negationFormat === 'parentheses') {
+        hasNegation = value[0] === '(' && value[value.length - 1] === ')';
```

Comment:
This logic is duplicate. Can we move it to util, and use that. 
```js
function getNegationMeta(value, negationFormat){
  let hasNegation;
  if (negationFormat === 'parentheses') {
    hasNegation = value[0] === '(' && value[value.length - 1] === ')';
  } else {
    hasNegation = value[0] === '-';
  }
  return {
     hasNagation,
     negationSuffixLegth: negationFormat === 'parentheses' && hasNegation ? 1 : 0,
  };
}
```
And this logic can be simple
```js
const {hasNagation, negationSuffixLegth} = getNegationMeta(value, negationFormat);
```